backend/server/middleware/authenticator.go

```diff
@@ -83,7 +82,7 @@ func Policy(ctx context.Context) policy.Policy {
 	return p
 }
 
-func AuthenticateAppAndInjectCtx(db *database.Connection) mux.MiddlewareFunc {
+func AuthenticateAppAndInjectCtx(db *database.Connection) func(http.Handler) http.Handler {
```

Comment:
maybe it's worth defining a new type called MiddlewareFunc, which is the same as `mux.MiddlewareFunc` and `func(http.Handler) http.Handler`

 But if it's just these 3 uses, maybe it's not that valuable, as it's really just a way to keep the types simpler.

Reply:
if it was just two cases I don't think I would, but three is enough in my book to create a type

Reply:
I added this to the bottom of authenticator.go - where would you put it? I googled around but didn't see any convention re where to put misc types

Reply:
Two perspectives:

* A lot of times I see types put at the top of a file where it gets used most. Or, if it'll have methods attached to it, then its own file. I think this works well some of the time, but there are times when you really just want to group data together, or create an alias (as we're doing here), and then i'm not sure it makes sense.
* Sometimes, for kind of miscellaneous types, I like to create a `types.go` file and put them all in there (in the same package), kind of mirroring what I've seen done in typescript. For smaller types _I_ think this helps, but to each their own.

I guess a 3rd perspective: You should do whatever you think is best. Putting this where it is now is probably fine, and can be moved later if someone has another idea about how it should be structured.

Reply:
yeah I might go with the TS route!